SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -0,0 +1,226 @@
+  } else if constexpr (BaseVal == init_val::max_half) {
+    return value<DataT>::max() / 2;
+  } else if constexpr (BaseVal == init_val::neg_inf) {
+    return -value<DataT>::inf();
+  } else if constexpr (BaseVal == init_val::nan) {
+    return value<DataT>::nan();
+  } else {
+    static_assert(BaseVal != BaseVal, "Unexpected base value value");
+  }
+}
+
+// Struct that will be used for invocating of simd constructor.
+template <init_val BaseVal, init_val Step> struct call_simd_struct {
+  template <typename DataT, int NumElems, typename TestCaseT>
+  esimd_tests::shared_vector<DataT>
+  call_simd(sycl::queue &queue, const std::vector<DataT> &ref_data) {
+    esimd_tests::shared_vector<DataT> result{NumElems,
+                                             shared_allocator<DataT>{queue}};
+    esimd_tests::shared_vector<DataT> shared_ref_data{
+        ref_data.begin(), ref_data.end(), shared_allocator<DataT>{queue}};
+    const auto ref = ref_data.data();
+    DataT step_value{get_base_value<DataT, Step>()};
+    DataT base_value{get_base_value<DataT, BaseVal>()};
+    queue.submit([&](sycl::handler &cgh) {
+      const auto ref = ref_data.data();
+      auto out = result.data();
+      cgh.single_task<
+          kernel_for_fill<DataT, NumElems, TestCaseT, BaseVal, Step>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            TestCaseT::template call_simd_ctor<DataT, NumElems>(
+                base_value, step_value, out);
+          });
+    });
+
+    return result;
+  }
+};
+
+// Constructing verbose failure message.
+template <int NumElems>
+void on_failure(const std::string &step_value, const std::string &base_value,
+                const std::string &elem_from_ref_data,
+                const std::string &elem_from_result_data,
+                const std::string &data_type,
+                const std::string &test_case_description) {
+  log::fail<NumElems>("Simd by " + test_case_description +
+                          " failed, retrieved: " + elem_from_result_data +
+                          ", expected: " + elem_from_ref_data +
+                          ", with step value: " + step_value +
+                          ", with base value: " + base_value,
+                      data_type);
+}
+
+// Struct for verifying test result, failed test if a retrived value is not
+// equal to an expected value.
+template <init_val BaseVal, init_val Step> struct verify_obtained_result {
+  template <typename DataT, int NumElems>
+  bool verify_test_result(const std::vector<DataT> &ref_data,
+                          const esimd_tests::shared_vector<DataT> &result_data,
+                          const std::string &data_type,
+                          const std::string &test_case_description) {
+    bool pass{true};
+    DataT step_value{get_base_value<DataT, Step>()};
+    DataT base_value{get_base_value<DataT, BaseVal>()};
+    size_t zero_elem{0};
+
+    if (!are_bitwise_equal(ref_data[zero_elem], base_value)) {
```

Comment:
Is `ref_data` a reference data? Why do we need to check the first item from the `ref_data`, the rest of items from the `result_data`?

Reply:
now we compare value from `result` vector with expected value, I've done it [here](https://github.com/intel/llvm-test-suite/pull/598/commits/652dcd3b7c72dceecf81bb7bd486398d8664c24d#diff-7fc90b589428e6e89076f07f5ff280a217b7f5c8e836bfe6c742734825074ef4R232)